src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2007,6 +2008,50 @@ protected override void HandleSetFunctionBreakpointsRequestAsync(IRequestRespond
             responder.SetResponse(response);
         }
 
+        protected override void HandleCompletionsRequestAsync(IRequestResponder<CompletionsArguments, CompletionsResponse> responder)
+        {
+            if (!m_isStopped)
+            {
+                responder.SetError(new ProtocolException("Failed to handle CompletionsRequest", new Message(1105, AD7Resources.Error_TargetNotStopped)));
+                return;
+            }
+
+            IDebugStackFrame2 frame = null;
+            int? frameId = responder.Arguments.FrameId;
+            if (frameId != null)
+                _ = m_frameHandles.TryGet(frameId.Value, out frame);
+
+            string command = responder.Arguments.Text;
+            if (command[0] == '`')
+                command = command.Substring(1);
+            else if (command.StartsWith("-exec", StringComparison.Ordinal))
+                command = command.Substring(6);
```

Comment:
If the command doesn't start with ``` or `-exec` don't you want to not call AutoComplete?

Reply:
Well that's the question. As I said if we talk about the vscode debug console and gdb nobody expects expression completion. If the completion is also used for Watches (spec doesn't talk about that) then that's something else.
And I guess for VS Immediate Window you'd expect expressions.

Reply:
If it doesn't start with ``` or `-exec` then it is an expression. Since GDB doesn't have expression auto complete, I think we need to just return an empty match list without calling the engine.